src/main/java/query/QueryExecution.java

```diff
@@ -0,0 +1,211 @@
+      AtomicReference<Exception> errorRef = new AtomicReference<>();
+      AtomicReference<ExplainResponse> explainRef = new AtomicReference<>();
+      ResponseListener<QueryResponse> queryListener =
+          new ResponseListener<>() {
+            @Override
+            public void onResponse(QueryResponse response) {
+              System.out.println("Execute Result: " + response);
+              executeRef.set(response);
+              latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              System.out.println("queryExecution Execution Error: " + e);
+              errorRef.set(e);
+              latch.countDown();
+            }
+          };
+
+      ResponseListener<ExplainResponse> explainListener =
+          new ResponseListener<>() {
+            @Override
+            public void onResponse(ExplainResponse response) {
+              System.out.println("Explain response: " + response);
+              explainRef.set(response);
+              latch.countDown();
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              System.out.println("queryExecution Explain Error: " + e);
+              errorRef.set(e);
+              latch.countDown();
+            }
+          };
+
+      // Check if this is an explain query
+      boolean isExplainQuery = query.trim().toLowerCase().startsWith("explain");
+
+      if (isPPL) {
+        System.out.println("Executing PPL query...");
+        // For explain queries, set the path to "/_explain"
+        String path = isExplainQuery ? "/_explain" : "/_plugins/_ppl";
+        PPLQueryRequest pplRequest = new PPLQueryRequest(query, new JSONObject(), path, "");
+
+        if (isExplainQuery) {
+          System.out.println("Calling pplService.explain()");
+          pplService.explain(pplRequest, explainListener);
+        } else {
+          System.out.println("Calling pplService.execute()");
+          pplService.execute(pplRequest, queryListener, explainListener);
+        }
+      } else {
+        System.out.println("Executing SQL query...");
+
+        if (isExplainQuery) {
+          // Remove "explain" prefix
+          String actualQuery = query.substring(7).trim();
+          System.out.println("SQL explain query for: " + actualQuery);
+
+          String path = "/_explain";
+          SQLQueryRequest sqlRequest = new SQLQueryRequest(new JSONObject(), actualQuery, path, "");
+
+          System.out.println("Calling sqlService.execute() with explain path");
+          sqlService.execute(sqlRequest, queryListener, explainListener);
+        } else {
+          // Regular SQL query
+          String path = "/_plugins/_sql";
+          SQLQueryRequest sqlRequest = new SQLQueryRequest(new JSONObject(), query, path, "");
+
+          System.out.println("Calling sqlService.execute()");
+          sqlService.execute(sqlRequest, queryListener, explainListener);
+        }
+      }
```

Comment:
let's refactor this code with design patterns, by learning Strategy Pattern and Factory Pattern.